src/components/Images/Images.js

```diff
@@ -1,30 +1,151 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import * as Cloudinary from 'cloudinary-react'
+import cloudinary from 'cloudinary-core'
+import useRequired from '../../hooks/useRequired.js'
+import useInvalid from '../../hooks/useInvalid.js'
+import { Media } from '../Media/Media'
+import uuidv4 from 'uuid/v4'
+import lazysizes from 'lazysizes'
+import styles from './Images.module.scss'
 
 // https://cloudinary.com/documentation/image_transformation_reference
-
 export const CLOUDINARY_CLOUD_NAME = 'getethos'
+const cld = new cloudinary.Cloudinary({
+  cloud_name: CLOUDINARY_CLOUD_NAME,
+})
+
+const mediaBreakpoints = [
+  Media.BREAKPOINTS.DESKTOP_RANGE_START,
+  Media.BREAKPOINTS.LAPTOP_RANGE_START,
+  Media.BREAKPOINTS.TABLET_RANGE_START,
+  Media.BREAKPOINTS.PHONE_RANGE_END,
+]
+
+export const CloudinaryImage = ({
+  publicId,
+  className,
+  alt,
+  width,
+  height,
+  crop,
+  ...rest
+}) => {
+
+  // Verify that all required props were supplied
+  const [includesRequired] = useRequired(['publicId', 'alt'])
+  const allRelevantProps = Object.assign({}, rest, {
+    publicId,
+    alt,
+    width,
+    height,
+  })
+  includesRequired(allRelevantProps)
+
+  // Verify that no invalid props were supplied
+  const [includesInvalid] = useInvalid(
+    Object.keys(CloudinaryImage.PUBLIC_PROPS)
+  )
+  includesInvalid(rest)
+
+  const { WEBP, JP2, JPEG, SVG } = CloudinaryImage.IMAGE_FILE_TYPES
+  const baseImageSettings = {
+    quality: 'auto:eco',
+    crop: crop,
+    secure: true,
+    flags: ['progressive:semi'],
+  }
+  let imageClasses = ['lazyload', className]
+  let reverseWidth, reverseHeight
+  width && (reverseWidth = width.slice().reverse())
```

Comment:
This code is setting `reverseWidth` to `width.slice().reverse()` if `width` is defined, correct? It is a little confusing so using `if` blocks may be better

Reply:
I agree here. I'd say unless we're in JSX we should shoot for readability first; especially since we have an assignment happening

Reply:
I needed to dupe width then set it to be the reversed version. Ill move it into an if statement